### ISSUE TITLE
Inconsistent CEPA(0) Definitions Across Psi

### ISSUE THREAD

**Issue by JonathonMisiewicz** [psi4/psi4#1872]
For the `fnocc` modules, [`cepa(0)` means `lccsd`](https://github.com/psi4/psi4/blob/master/psi4/driver/procrouting/proc.py#L4655-L4660). I have only ever heard of `cepa(0)` meaning `lccd`. It is `lccd` in the `dct` module, it is `lccd` in the `occ` module, and it is `lccd` in the `dfocc` module. Can we make this consistent, or at least print a warning?

CEPA(0) is very important to my own research, and I had only ever heard of it as `lccd`. After doing some more digging, I've come across some sources that equate CEPA(0) and `lccsd`, but it seems to me to be a minority position. 

Making this even more complicated, all requests for `cepa(0)` go to `fnocc`. `occ` and `dfocc` require `lccd`. `dct` also requires `cepa0`, but as a keyword rather than the energy name.

Obligatory ping of @edeprince3 for `fnocc` issues and @loriab for defining methods consistently across modules and @jturney because this has been causing problems at CCQC.

**Comment by loriab** [comment]
Here's a few more refs and the original discussion http://forum.psicode.org/t/naming-considerations-for-cepa-0/156 . And more ways it can confuse ppl http://forum.psicode.org/t/cepa-0-singles/1525/11 .

My main consideration is that `energy("cepa(0)")` is running the same method in all modules. That's what's currently forced by cepa(0) directing all traffic to fnocc (since that's the only module that can do lccsd). `lccd` properly manages direction to fnocc/occ. If the opinion of `cepa0 --> lccsd` has shifted over the years to `cepa0 --> lccd`, that's ok with me, and with a lot of warnings, we can shift the definition. Or, if you'd like `cepa0` to always raise an error and demand sd/d distinction, that's ok, too. Or maybe dct could adjust to the current scheme.

Basically, I think the current way is sane. But then I don't use cepa. And the current setup has shown itself to be confusing to new (thread above) and experienced (AED, Jonathon) practitioners. So strong feelings from multiple other parties could overcome my symlink.

**Comment by JonathonMisiewicz** [comment]
...Well, thanks Molpro. We have a legitimate ambiguity after all.

Okay, new plan:
* `cepa(0)` stays `lccsd` for `energy`, `gradient`, etc. purposes.
* Any time `cepa(0)` appears in an output file, we clarify what we're talking about. Explicitly saying "Linearized CCD" or "Linearized CCSD" is ideal. If we need to abbreviate, I've seen some Bartlett papers where they use `LinCCD`. Crawford has a good point about `LCCD` referring to localization. If we try to eliminate `cepa(0)` meaning `lccd` entirely, we create havoc with Bozkaya's papers. If we have a logging module, we print out a warning there too.
* `cepa` is eliminated from `dct`. Andy put that in there years ago to facilitate comparison between Kutzelnigg's first stab at DCT and LCCD. Without getting into the theory weeds, that's  not a useful comparison now, and it's going to be an even less useful comparison after I've published my next two papers. It's an awkward addition given the DCT logic. It's easiest to get rid of it and direct users to `occ`, `dfocc`, or `fnocc`.

**Comment by loriab** [comment]
> ...Well, thanks Molpro. We have a legitimate ambiguity after all.

:-)

> cepa(0) stays lccsd for energy, gradient, etc. purposes.

sounds good -- can avoid deprecation warnings

> Any time cepa(0) appears in an output file, we clarify what we're talking about. Explicitly saying "Linearized CCD" or "Linearized CCSD" is ideal. If we need to abbreviate, I've seen some Bartlett papers where they use LinCCD. Crawford has a good point about LCCD referring to localization. If we try to eliminate cepa(0) meaning lccd entirely, we create havoc with Bozkaya's papers. If we have a logging module, we print out a warning there too.

* overall sounds good to add very specific text clarifying CCD/CCSD. I'm less concerned about linearized/local since psi has none of the latter.
* should try to keep output of cepa-0,1,2,etc. cohesive for fnocc
* I'd like to not disrupt the qcvars, presently e.g., `LCCD OPPOSITE-SPIN CORRELATION ENERGY`, `LCCSD TOTAL ENERGY`, but I'm not adamant.
* I'm less concerned about cohesiveness with UB papers. After all, once upon a time, the occ/dfocc code was "using lccd for non-orbital-optimized and ocepa0 for orbital-optimized". Any firm connections btwn code and paper can be cemented by pasting some DOIs. IMO, psi should try to be cohesive from the face, not from the lit perspective.

> cepa is eliminated from dct. Andy put that in there years ago to facilitate comparison between Kutzelnigg's first stab at DCT and LCCD. Without getting into the theory weeds, that's not a useful comparison now, and it's going to be an even less useful comparison after I've published my next two papers. It's an awkward addition given the DCT logic. It's easiest to get rid of it and direct users to occ, dfocc, or fnocc.

sounds good.

**Comment by JonathonMisiewicz** [comment]
I'll make the remaining `cepa0` changes after #1850 comes in. It seems prudent to avoid _two_ PRs standardizing things in `occ`, `dfocc`, and `fnocc` at once. The issue can then close.
